### PR TITLE
Fix #770 Use id instead of .extension

### LINKED ISSUE #770: [Bug] Can't edit IDN ccTLD
### Describe the bug

hey,

in a past fossbilling version, i added an idn top level domain ( (IDN ccTLD) .ελ - details in https://en.wikipedia.org/wiki/.gr#Alternative_top_domain )  
when trying to edit today with fully upgraded fossbilling (0.2.10), got a 404 : https://url/admin/servicedomain/tld/ελ

so, deleted tld, and readded (without errors), but still can't edit it. same 404. 

nothing on data/log/php_error.log : 
`[01-Feb-2023 23:20:35 timezone] Page /servicedomain/tld/ελ not found`

apache log : 
`[01/Feb/2023:23:20:35] GET /admin/servicedomain/tld/%CE%B5%CE%BB HTTP/2.0" 404 17134 "https://url/admin/servicedomain`

exactly the same applies for .ευ (https://en.wikipedia.org/wiki/.eu#Greek_domain) and any other IDN ccTLD i've tried. (even with cyrillic, not just greek characters). eg. .ею (https://en.wikipedia.org/wiki/.eu#Cyrillic_domain)

running on virtualmin host . Debian 11, apache2.4, Mariadb 10.5, php8.1

thanks in advance,

### How to reproduce

1. Go to /admin/servicedomain
2. Add any IDN ccTLD and save
3. Try to edit added IDN ccTLD

### Expected behavior

be able to edit ccTLDs/details. As TLDs prices/offers keep changing it doesn't seem wise to have to delete TLD + re-add each time. also not sure how/if such "workaround" would affect existing customers/orders (?)

### Screenshots

_No response_

### FOSSBilling version

0.2.10 (latest stable)

### Module version

_No response_

### Device

Desktop

### Information

- Device : PC
- OS : Debian testing amd64
- Browser : Firefox 
- Version : 109

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jaapmarcus**
Replaces: https://github.com/FOSSBilling/FOSSBilling/pull/771

### CODE REVIEW

**Review by BelleNottelling — APPROVED** [review]
Seems good to me, I don't know if we'd want to keep the old route or not. I can't think of any situation where it would be needed, so I think we should be good without it and the PR is good as-is

**Comment by jaapmarcus** [comment]
> Seems good to me, I don't know if we'd want to keep the old route or not. I can't think of any situation where it would be needed, so I think we should be good without it and the PR is good as-is

That was my first idea it should be needed kept the other "api" calls in place as they are using other locations for order and so on.

**Comment by BelleNottelling** [comment]
> > Seems good to me, I don't know if we'd want to keep the old route or not. I can't think of any situation where it would be needed, so I think we should be good without it and the PR is good as-is
> 
> That was my first idea it should be needed kept the other "api" calls in place as they are using other locations for order and so on.

Can't we keep that route then, and add the fixed regex from https://github.com/FOSSBilling/FOSSBilling/pull/771 for the most fool-proof fix to the issue?

File: src/modules/Servicedomain/Api/Admin.php [hunk 1]
```diff
@@ -197,6 +197,22 @@ public function tld_get($data)
         return $this->getService()->tldToApiArray($model);
     }
 
+    public function tld_get_id($data)
+    {
+        var_dump($data);
```
Comment (BelleNottelling) [inline src/modules/Servicedomain/Api/Admin.php:202]: Leftover debug.

**Review by John-S4 — APPROVED** [review]

**Pull request merged**